demo/clojure/echo.clj

```diff
@@ -0,0 +1,106 @@
+;;#!/usr/bin/env bb
+
+(ns maelstrom.echo
+  (:gen-class)
+  (:require
+    [cheshire.core :as json]
+    [clojure.java.io :as io]))
+
+
+;;;;;;;;;;;;;;;;;;; Util functions ;;;;;;;;;;;;;;;;;;;
+
+;;;;;; Input pre-processing functions ;;;;;;
+
+(defn- read-file
+  "Read a file into a vector of strings.
+  This is used for local testing in repl"
+  [f]
+  (with-open [rdr (io/reader (io/input-stream f))]
+    (reduce conj [] (line-seq rdr))))
+
+
+(defn- process-stdin
+  "Read lines from the stdin and calls the handler"
+  [handler]
+  (doseq [line (line-seq (java.io.BufferedReader. *in*))]
+    (handler line)))
+
+
+(defn- parse-json
+  "Parse the received input as json"
+  [input]
+  (try
+    (json/parse-string input true)
+    (catch Exception e
+      nil)))
+
+
+;;;;;; Output Generating functions ;;;;;;
+
+(defn- generate-json
+  "Generate json string from input"
+  [input]
+  (json/generate-string input))
+
+
+(defn- printerr
+  "Print the received input to stderr"
+  [input]
+  (binding [*out* *err*]
+    (println input)))
+
+
+(defn- printout
+  "Print the received input to stdout"
+  [input]
+  (println input))
+
+
+(defn reply
+  ([src dest body]
+   {:src src
+    :dest dest
+    :body body}))
+
+
+(def node-id (atom ""))
+
+
+(defn- process-request
+  [input]
+  (let [body (:body input)
+        r-body {:msg_id (rand-int 100)
```

Comment:
Is there a reason why you're generating random message IDs here? They're supposed to be unique--I think collision is fine in this case, but I can't guarantee that's the case always.

Reply:
Will change to a counter